prescient/simulator/prescient.py

```diff
@@ -11,127 +11,103 @@
+        # For safety, in case we re-run Prescient
+        # again in the same Python script
+        # (For the same reason, we don't accept options
+        #  in __init__ above.)
+        prescient.plugins.internal.clear_plugins()
+        prescient.simulator.config.clear_prescient_config()
+
+        if 'config_file' in options:
+            config_file = options.pop('config_file')
+            if options:
+                raise RuntimeError(f"If using a config_file, all options must be specified in the configuration file")
+            script, config_options = runner.parse_commands(config_file)
+            if script != 'simulator.py':
+                raise RuntimeError(f"config_file must be a simulator configuration text file, got {script}")
+            options = parse_args(args=config_options)
+
+        elif 'plugin' in options:
+            # parse using the Config
+            plugin_options = self.CONFIG({ 'plugin':options['plugin'] })
+            for plugin in plugin_options.plugin:
+                # importing the plugin will update
+                # both the global config and the
+                # global plugin registration
+                import_file(plugin)
+            # to reload after its changed by a plugin
+            from .config import PrescientConfig
+            options = PrescientConfig(options)
         else:
-            errmsg = None
-            try:
-                simulator = create_prescient(options)
-                ans = simulator.simulate(options)
-            except ValueError:
-                err = sys.exc_info()[1]
-                errmsg = 'VALUE ERROR: %s' % err
-            except KeyError:
-                err = sys.exc_info()[1]
-                errmsg = 'KEY ERROR: %s' % err
-            except TypeError:
-                err = sys.exc_info()[1]
-                errmsg = 'TYPE ERROR: %s' % err
-            except NameError:
-                err = sys.exc_info()[1]
-                errmsg = 'NAME ERROR: %s' % err
-            except IOError:
-                err = sys.exc_info()[1]
-                errmsg = 'I/O ERROR: %s' % err
-            #Can't find ConverterError? Commenting out for now.
-            #except ConverterError:
-                #err = sys.exc_info()[1]
-                #errmsg = 'CONVERSION ERROR: %s' % err
-            except RuntimeError:
-                err = sys.exc_info()[1]
-                errmsg = 'RUN-TIME ERROR: %s' % err
-            #pyutilib no attribute named common? Commenting out for now.
-            #except pyutilib.common.ApplicationError:
-                #err = sys.exc_info()[1]
-                #errmsg = 'APPLICATION ERROR: %s' % err
-            except Exception:
-                err = sys.exc_info()[1]
-                errmsg = 'UNKNOWN ERROR: %s' % err
-                traceback.print_exc()
-
-            if errmsg is not None:
-                sys.stderr.write(errmsg + '\n')
-
-    return ans
+            options = self.CONFIG(options)
```

Comment:
Each of the three paths takes a different approach to returning a populated `options` object. The file-based approach returns an argparse.Namespace, the plugin code comes from PrescientConfig, and the "normal" path is based on self.CONFIG (which should be the same as PrescientConfig, but as John pointed out, after a reset it is not). At the very least, IMO they should all return the same data type.